internal/vault/credentials/aws.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package credentials
+
+import (
+	"context"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/hashicorp/go-secure-stdlib/awsutil"
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+	"github.com/hashicorp/vault-secrets-operator/internal/consts"
+)
+
+const (
+	AWSAnnotationRole            = "eks.amazonaws.com/role-arn"
+	AWSAnnotationAudience        = "eks.amazonaws.com/audience"
+	AWSAnnotationTokenExpiration = "eks.amazonaws.com/token-expiration"
+	AWSDefaultAudience           = "sts.amazonaws.com"
+	AWSDefaultTokenExpiration    = int64(86400)
+)
+
+var _ CredentialProvider = (*AWSCredentialProvider)(nil)
+
+type AWSCredentialProvider struct {
+	authObj           *secretsv1alpha1.VaultAuth
+	providerNamespace string
+	uid               types.UID
+}
+
+func (l *AWSCredentialProvider) GetNamespace() string {
```

Comment:
Doesn't matter, but OOI why `l`?

Reply:
I don't really know; it's that way in the other providers too.